src/Silicon.cpp

```diff
@@ -1372,7 +1372,9 @@ namespace galsim {
             // Get the location where the photon strikes the silicon:
             double x0 = photonsX[i]; // in pixels
             double y0 = photonsY[i]; // in pixels
+#ifdef DEBUGLOGGING
```

Comment:
These ifdef guards are superfluous.  Please remove them.  (xdbg is already a no op when DEBUGLOGGING is not defined.)

Reply:
Without the guards, the GPU build fails on both gcc and nvc++ (though not clang). My guess would be that the GPU compiler sees std::cerr and throws an error before realising that it's unreachable. I think we either need the guards or we'll need to modify the implementation of xdbg so that disabling it works differently.

Reply:
How about changing `std:cerr` to `*dbgout` and have an `extern std::ostream* dbgout` in the no DEBUGLOGGING block.  You might need to define this variable somewhere too in a .cpp file, but maybe not if the compiler always converts it to a no op in the .o files.

Reply:
That solves the problem.